rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -19,7 +19,7 @@ trait Substitute[M[_], A] {
 }
 
 object Substitute {
-  private[interpreter] def charge[A: Chargeable, M[_]: Sync: _cost: _error](
+  private[interpreter] def charge[A: Chargeable, M[_]: Sync: _cost: _costWriter: _error](
```

Comment:
We must not thread a separate `_costWriter` capability. Instead, we should redefine `_cost` to include this capability.

Reply:
The way this should work, is:

we'd have a standard `_cost` instance that does no logging
we'd have a `_cost` instance that does the logging, and requires `FunctorTell[F, Cost]` for **creating** (not using!) it.

Reply:
How do you want to "include" `FunctorTell` in a `MonadState`?